arelle/plugin/validate/NL/rules/nl_kvk.py

```diff
@@ -429,6 +429,37 @@ def rule_nl_kvk_3_3_1_3 (
         )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NL_INLINE_2024
+    ],
+)
+def rule_nl_kvk_3_5_2_1(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    NL-KVK.3.5.2.1: Each tagged text fact MUST have the ‘xml:lang’ attribute assigned or inherited.
+    """
+    factsWithoutLang = set()
+    for fact in val.modelXbrl.facts:
+        if (fact is not None and
```

Comment:
Out of curiosity, why `set()` rather than `[]`?  I would not expect `modelXbrl.facts` to have duplicates.

Out of curiosity, can `modelXbrl.facts` actually contain `None`?

Reply:
I updated to list rather than set. As far as `modelXbrl.facts` containing None, I dont have any examples to show you, but feel better about checking that prior just for safety sake.

Reply:
I appreciate the need to always guard against None in Arelle. That said, the modelXbrl.facts collection should never contain None values. If it does, we’ll end up throwing exceptions throughout the codebase.